server/src/dashboards/dashboards.service.ts

```diff
@@ -0,0 +1,93 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common'
+import { InjectRepository } from '@nestjs/typeorm'
+import { User } from 'src/auth/user.types'
+import { CategoriesService } from './categories.service'
+import { Dashboard } from './dashboard.entity'
+import { DashboardRepository } from './dashboard.repository'
+import { CreateDashboardDto, UpdateDashboardDto } from './dashboards.dto'
+
+@Injectable()
+export class DashboardsService {
+  constructor(
+    @InjectRepository(DashboardRepository)
+    private dashboardRepository: DashboardRepository,
+    @Inject(forwardRef(() => CategoriesService))
+    private categoriesService: CategoriesService,
+  ) {}
+
+  /**
+   * Returns all the dashboards for a user
+   *
+   * @param user
+   */
+  async getAllDashboards(user: User): Promise<Dashboard[]> {
+    const dashboards = await this.dashboardRepository.find({
+      order: { createdAt: 'DESC' },
+      where: { uid: user.uid },
+    })
+    return dashboards
+  }
+
+  /**
+   * Returns the dashboard data
+   *
+   * @param id - id of the dashboard user wants to fetch
+   * @param user
+   */
+  async getDashboard(id: number, user: User): Promise<Dashboard> {
+    const dashboard = await this.dashboardRepository.findOne({
+      id,
+      uid: user.uid,
+    })
+    return dashboard
```

Comment:
Add checks if dashboard is present or not

```
if (!dashboard) {
      throw new NotFoundException('dashboard not found')
}
```

Reply:
Similarly check all the service methods you had created, `chart`, `dashboard-chart`, `dashboard`, `category`.